.github/workflows/aws-eks-test.yml

```
@@ -56,11 +56,20 @@ jobs:
 
       - name: report failure to slack
         if: failure()
-        id: slack
+        id: slack-failure
         uses: slackapi/slack-github-action@v1.23.0
         with:
-          channel-id: '${{ secrets.SLACK_BOT_CHANNEL }}'
-          slack-message: "Nightly EKS test failed: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+          channel-id: '${{ secrets.SLACK_BOT_FAILURE_CHANNEL }}'
+          slack-message: "❌ Nightly EKS test failed: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+
+       - name: report success to slack
+        id: slack-success
```

```suggestion
        if: success()
        id: slack-success
```
Will this not produce false positive "success" messages without checking the status?

---

Great question! [GitHub defaults to having this behavior unless otherwise specified](https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions). So, if a previous job fails then it will not run the next one. Thus, if a step (our test step in this case) fails then subsequent steps will not run which is why we need `if: failure()` and `if: always()`. In the case that it passes, this step will to run due to its default `if: success()` but will not if a failure occurs in a previous step.

---

Thank you for that information!